### ISSUE TITLE
Invalid return after valid return generates no error

### ISSUE THREAD

**Issue by refi64** [python/mypy#364]
``` python
def f() -> str:
    return '1'
    return 1
```

In my opinion, this still should cause an error, not exit silently.

**Comment by ilevkivskyi** [comment]
> Maybe we should warn about dead code after a return statement (and break/continue)?

Now that we have `--warn-unreachable`, I think this can be closed.
